### PR TITLE
update tests for el9/upcoming (SOFTWARE-5337)

### PR DESCRIPTION
**Pull request by edquist**

### CODE REVIEW

File: parameters.d/osg36-el9.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  - opensciencegrid:master; 3.6; osg, epel-testing
+  - opensciencegrid:master; 3.6; osg-testing
+  - opensciencegrid:master; 3.6; osg > osg-testing
+  # - opensciencegrid:master; 3.5; osg, osg-upcoming > 3.6/osg
+  # - opensciencegrid:master; 3.6; osg, osg-upcoming
+  # - opensciencegrid:master; 3.6; osg-testing, osg-upcoming-testing
+  # - opensciencegrid:master; 3.6; osg > osg-testing, osg-upcoming-testing
+  # - opensciencegrid:master; 3.5; osg, osg-upcoming > 3.6/osg-minefield
+  # - opensciencegrid:master; 3.6; osg-minefield
+  # - opensciencegrid:master; 3.6; osg > osg-minefield
```
Comment (matyasselmeci) [inline parameters.d/osg36-el9.yaml:31]:
```suggestion
  # - opensciencegrid:master; 3.6; osg, osg-upcoming
  # - opensciencegrid:master; 3.6; osg-testing, osg-upcoming-testing
  # - opensciencegrid:master; 3.6; osg > osg-testing, osg-upcoming-testing
  - opensciencegrid:master; 3.6; osg-minefield
  - opensciencegrid:master; 3.6; osg > osg-minefield
```
3.5 will never support EL9. Also we should uncomment the minefield tests for now until we have more stuff in testing.

File: parameters.d/osg36-upcoming-el9.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  # - opensciencegrid:master; 3.6; osg, epel-testing
+  # - opensciencegrid:master; 3.6; osg-testing
+  # - opensciencegrid:master; 3.6; osg > osg-testing
+  - opensciencegrid:master; 3.5; osg, osg-upcoming > 3.6/osg
+  - opensciencegrid:master; 3.6; osg, osg-upcoming
+  - opensciencegrid:master; 3.6; osg-testing, osg-upcoming-testing
+  - opensciencegrid:master; 3.6; osg > osg-testing, osg-upcoming-testing
+  # - opensciencegrid:master; 3.5; osg, osg-upcoming > 3.6/osg-minefield
+  # - opensciencegrid:master; 3.6; osg-minefield
+  # - opensciencegrid:master; 3.6; osg > osg-minefield
```
Comment (matyasselmeci) [inline parameters.d/osg36-upcoming-el9.yaml:31]:
```suggestion
  # - opensciencegrid:master; 3.6; osg
  # - opensciencegrid:master; 3.6; osg, epel-testing
  # - opensciencegrid:master; 3.6; osg-testing
  # - opensciencegrid:master; 3.6; osg > osg-testing
  - opensciencegrid:master; 3.6; osg, osg-upcoming
  - opensciencegrid:master; 3.6; osg-testing, osg-upcoming-testing
  - opensciencegrid:master; 3.6; osg > osg-testing, osg-upcoming-testing
  - opensciencegrid:master; 3.6; osg-minefield
  - opensciencegrid:master; 3.6; osg > osg-minefield
```
Same as above.

File: parameters.d/osg36-el9.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+package_sets:
+  #### Required ####
+  # label - used for reporting, should be consistent across param files
+  # packages - list of packages to install in the test run
+  #### Optional ####
+  # selinux - enable SELinux for the package set, otherwise Permissive mode (default: True)
+  # osg_java - Pre-install OSG java packages (default: False)
```
Comment (matyasselmeci) [inline parameters.d/osg36-el9.yaml:39]: wow this stuff is old.
It's not in scope for this PR but in the future we should go through and remove all the unused stuff like this...

**Review by brianhlin — APPROVED** [review]
Mostly LGTM. Let's see what a manual test run looks like before we merge into master.

**Review by matyasselmeci — APPROVED** [review]
Minor suggestions; pre-approving.

File: parameters.d/osg36-el9.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  # [<Github account>:<osg-test branch>;] <OSG ver>; <REPO 1, REPO 2...REPO N> [> <Update OSG ver>/<Update REPO 1, REPO 2...REPO N>]
+  # Example:
+  # Run osg-test (from 3.2-minefield) with packages from 3.2-release and 3.2-testing that are then upgraded to
+  # 3.3-testing and 3-3-upcoming-testing:
+  # 3.2; osg, osg-testing > 3.3/osg-testing, osg-upcoming-testing
+  ###################
+  - opensciencegrid:master; 3.6; osg
+  - opensciencegrid:master; 3.6; osg, epel-testing
+  - opensciencegrid:master; 3.6; osg-testing
+  - opensciencegrid:master; 3.6; osg > osg-testing
```
[the remark this answers is not in the document]
Comment (edquist) [reply parameters.d/osg36-el9.yaml:24]: Ah right no upgrade tests yet, good catch

**Comment by edquist** [comment]
> Let's see what a manual test run looks like before we merge into master.

Alright kicked.

**Comment by edquist** [comment]
@brianhlin -

Basically all the INSTALLs failed for EL9:

https://osg-sw-submit.chtc.wisc.edu/tests/20230131-1142/packages.html

Stuff like:

```
No match for argument: osg-ce-condor
```

I think we just have more el9 packages to build.

**Pull request merged**